code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -24,7 +24,27 @@
 	var/isbroken = 0
 	light_range = 5
 	light_color = LIGHT_COLOR_RED
-	var/obj/item/wepon = null
+	var/last_check
+
+/obj/structure/cult/pylon/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/cult/pylon/Destroy()
+	..()
+	processing_objects.Remove(src)
+
+/obj/structure/cult/pylon/process()
+	if(!isbroken && world.time > last_check + 3 SECONDS)
+		var/list/can_see = view(src, 3)
+		last_check = world.time
+		for(var/mob/living/simple_animal/construct/C in can_see)
```

Comment:
and oview and range and orange and hearers and ohearers and viewers and oviewers.
any of those (and possibly some more) have the optimisation.

@PJB3005 I think they were saying that if the pylon did something else in ```view(src, 3)``` whether or not they should use the same loop?